### PR TITLE
Render array-like data as an array.

### PR DESCRIPTION
**Pull request by AndrewKvalheim**
In case you find it useful for #35, I've been using this to imitate Firebase's [array rendering][arrays] behavior.

  [arrays]: https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html

### CODE REVIEW

File: test/unit/firebase.js [hunk 1]
```diff
@@ -36,6 +36,29 @@ describe('MockFirebase', function () {
     });
   });
 
+  describe('Array rendering', function () {
+    it('renders integer–keyed objects as arrays', function () {
```
Comment (soumak77) [inline test/unit/firebase.js]: Not all outputs are valid.  If the keys are not sequential, the returned data from firebase is an object.  See the comment I added https://github.com/soumak77/firebase-mock/issues/35#issuecomment-323888463

Comment (AndrewKvalheim) [reply test/unit/firebase.js]: Looks like it handles those examples correctly. Perhaps the name of the test is misleading? s/integer-keyed/array-like/?

Comment (soumak77) [reply test/unit/firebase.js]: Among others, this test is wrong:

`{input: {0: 'a', 2: 'c', 3: 'd', 4: 'e'}, output: ['a', null, 'c', 'd', 'e']}`

The output should be the same as the input.

**Comment by AndrewKvalheim** [comment]
Hmm, bitten by isaacs/github#443.

> @soumak77 commented on this pull request.
>
> In [test/unit/firebase.js](https://github.com/soumak77/firebase-mock/pull/37#discussion_r148896656):
>
> ```diff
> > @@ -36,6 +36,29 @@ describe('MockFirebase', function () {
>      });
>    });
>  
> +  describe('Array rendering', function () {
> +    it('renders integer–keyed objects as arrays', function () {
> ```
>
> Among others, this test is wrong:
>
> ```javascript
> {input: {0: 'a', 2: 'c', 3: 'd', 4: 'e'}, output: ['a', null, 'c', 'd', 'e']}
> ```
> 
> The output should be the same as the input.

I'm seeing:

```shell
$ firebase database:set --confirm '/sandbox' --data '{"0": "a", "2": "c", "3": "d", "4": "e"}' >/dev/null && firebase database:get '/sandbox'
["a",null,"c","d","e"]
```

**Comment by soumak77** [comment]
I wonder whether the command line handles arrays differently than the javascript library

**Comment by AndrewKvalheim** [comment]
Seems consistent with firebase@3.9.0 and 4.6.0, modulo the question of null vs. undefined:

```console
> var output; ref.set({0: 'a', 2: 'c', 3: 'd', 4: 'e'}).then(() => ref.once('value')).then(snapshot => output = snapshot.val()); null
null
> output
[ 'a', <1 empty item>, 'c', 'd', 'e' ]
> output[1]
undefined
```

**Comment by soumak77** [comment]
@AndrewKvalheim please fix the broken unit tests

**Comment by AndrewKvalheim** [comment]
Which tests does this break?

**Comment by soumak77** [comment]
It was breaking on older node versions.  I just removed those versions from the list of versions to test against.  Changes have been published with v1.1.6.

**Pull request merged**